code/datums/components/garrote.dm

```diff
@@ -0,0 +1,53 @@
+ 
+#define STANDARD_MULT 1
+#define NO_DAMAGE 0
+
+/datum/component/garrote
+	var/obj/item/weapon
+	var/mob/living/target
+	var/grab_speed_mult = STANDARD_MULT
+	var/damage = NO_DAMAGE
+
+/datum/component/garrote/Initialize(mob/living/targ, obj/item/wpn, _grab_speed_mult, _damage)
+
+	if(!isliving(parent) || !iscarbon(targ))
+		return COMPONENT_INCOMPATIBLE
+	
+	var/mob/living/attacker = parent
+
+	if(_grab_speed_mult)
+		grab_speed_mult = _grab_speed_mult
+	if(_damage)
+		damage = _damage
+	target = targ
+	weapon = wpn
+
+	if(HAS_TRAIT(attacker, TRAIT_PACIFISM))
+		to_chat(attacker, "<span class='warning'>You couldn't bring yourself to strangle someone...</span>")
+		return
+	
+	if(!targ.density && !iscarbon(targ))
```

Comment:
```suggestion
	if(!targ.density))
```
is there a reason you repeated the iscarbon check here?
also why not move this up to line 14, with the other check?